homeassistant/components/ecobee/config_flow.py

```diff
@@ -33,7 +33,11 @@ def __init__(self):
             return self.async_abort(reason="one_instance_only")
 
         errors = {}
-        stored_api_key = self.hass.data[DATA_ECOBEE_CONFIG].get(CONF_API_KEY)
+        stored_api_key = (
+            self.hass.data[DATA_ECOBEE_CONFIG].get(CONF_API_KEY)
+            if DATA_ECOBEE_CONFIG in self.hass.data
+            else None
```

Comment:
I think we should default to empty string to match the type in the config schema.
```suggestion
            else ""
```